include/acl_types.h

```diff
@@ -1346,6 +1350,10 @@ typedef struct {
   // it sees the CL_COMPLETE status for the first time, but it won't
   // change after that.
   cl_uint num_printf_bytes_pending;
+
+  // Indicate whether this operation is dumping printf buffer before the Kernel
+  // for debug purpose
+  int debug_dump_printf = 0;
```

Comment:
Would anything speak against making this a `bool`, since it only has two states?

Reply:
Never mind, the type `int` is aligned with the HAL API, correct?
https://github.com/intel/fpga-runtime-for-opencl/blob/67f7f880f7af5af0c1797c4dd2a70b4e27ba85ca/include/acl_hal.h#L65-L66